internal/sam/elog.go

```diff
@@ -93,36 +99,43 @@ func (eo *ElogOperation) setr(r []rune) {
 	copy(eo.r, r)
 }
 
-func (e *Elog) Replace(q0, q1 int, r []rune) {
+func (e *Elog) Replace(q0, q1 int, r []rune) error {
+	var err error = nil
 	if q0 == q1 && len(r) == 0 {
-		return
+		return err
 	}
 
 	eo := e.last()
 
 	// Check for out-of-order
 	if q0 < eo.q0 && !e.warned {
 		e.warned = true
-		warning(nil, Wsequence)
+		err = errors.New(Wsequence)
```

Comment:
Check err?

Reply:
Or alternately, why is the check so far from the assignment?

Reply:
I'm attempting to minimize changes to the original code, nothing was done with the warning prior to the method being returned previously, the logic here will change significantly following the use of undo.Buffer in a few PRs anyways.

Reply:
> So long as the "this" its referring too moved as well!
> […](#)
> On Fri, Aug 6, 2021 at 2:01 PM camsn0w ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In observableeditablebuffer.go <[#349 (comment)](https://github.com/rjkroege/edwood/pull/349#discussion_r684502455)>: > @@ -16,7 +18,15 @@ type ObservableEditableBuffer struct { currobserver BufferObserver observers map[BufferObserver]struct{} // [private I think] f *File - details *file.DiskDetails + elog sam.Elog + // TODO(rjk): Remove this when I've inserted undo.RuneArray. It was previously with the elog declaration in file.go so I moved it with elog over to OEB — You are receiving this because you commented. Reply to this email directly, view it on GitHub <[#349 (comment)](https://github.com/rjkroege/edwood/pull/349#discussion_r684502455)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAJ6RUGJ7OGW3EEG6FAYDODT3REULANCNFSM5BLNY66Q> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email> .

It was